core/consensus/common/chainedbft/smr/smr.go

```diff
@@ -603,9 +633,9 @@ func (s *Smr) UpdateValidateSets(validates []*cons_base.CandidateInfo) error {
 }
 
 // getReplicasURL return validates urls
-func (s *Smr) getReplicasURL() []string {
+func (s *Smr) getReplicasURL(validates []*cons_base.CandidateInfo) []string {
```

Comment:
If  use param to transmit `validates`. This function does not need to be a member function.